experiment/simulation/js/main.js

```diff
@@ -1 +1,83 @@
-//Your JavaScript goes in here
+let sleep = (ms) => {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+};
+
+let arrayRemove = (arr, value) => {
+  return arr.filter(function (ele) {
+    return ele != value;
+  });
+};
+
+let ids = ["long", "lcontainer"];
+
+let asyncMove = async (id, curPosition = 0, finalPosition = 1) => {
+  let path = document.getElementById(id);
+  let flags = [true, true];
+  while (true) {
+    let animation_speed = document.getElementById("water-flow").value;
+    animation_speed = animation_speed * 0.0001;
+    animation_speed = animation_speed == 0 ? 0.0001 : animation_speed;
+    if (curPosition > finalPosition) break;
+    curPosition += animation_speed;
+    if (id == "left-curve") {
```

Comment:
Please use === for comparison, it tends to catch more errors than == and shoul dbe the default comparison operator.